core/src/main/java/org/mule/api/debug/Debuggeable.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+
+package org.mule.api.debug;
+
+/**
+ * Specifies that the message processor that implements this interface can debug its internal execution.
+ *
+ * @since 3.8.0
+ */
+public interface Debuggeable
+{
+    /**
+     * Enables the debug mode for this message processor
+     *
+     * @param callback The callback to be used to interact with the mule debugger
+     * @return The debugging session
+     */
+    DebuggingSession enableDebug(DebuggerCallback callback);
+
+    /**
+     * If it is already enabled or not.
+     *
+     * @return True if it is enabled
+     */
+    boolean isDebugEnabled();
+
+    /**
+     * Disables the debugging mode.
+     *
+     */
+    void disableDebug();
```

Comment:
Should this javadoc explain that any stopped messages should continue execution?